be/src/runtime/buffer_control_block.cpp

```diff
@@ -119,7 +119,7 @@ Status BufferControlBlock::add_batch(std::unique_ptr<TFetchDataResult>& result)
 
     int num_rows = result->result_batch.rows.size();
 
-    while ((!_batch_queue.empty() && _buffer_rows > _buffer_limit) && !_is_cancelled) {
+    while ((!_mysql_batch_queue.empty() && _buffer_rows > _buffer_limit) && !_is_cancelled) {
```

Comment:
maybe we need another pr to set buffer limit from session variable. Then for arrow flight we may need change the session variable to get better performance.

Reply:
meaningful